### PR TITLE
Properties configs

### PR DESCRIPTION
**Pull request by gallyamb**
This branch includes following changes:
- add ability for passing parameters to player via properties file
- change player initializing strategy. Now we can pass class, that implements IConfig, to Player's initFromConfig method
- add insufficient parameters
- refactoring
- default scripts
- fix logging in player
- add dynamic Xvfb displays range
- likely that the user interface won't work (need some extra work)

### CODE REVIEW

File: player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    {
+        this.isAsyncRequestsCompletedScript = isAsyncRequestsCompletedScript;
+    }
+
+    @Override
+    public void loadDefaults()
+    {
+        // Please, sorry me for this hell, but otherwise it would look very ugly
+        // Here we initialize all uninitialized script fields with default script from resources
+        List<String> extensions = Arrays.asList(".groovy", ".js"); // It would be better, if extensions will be sorted by frequency in resources
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: Jesus. Sorry, but it must be rewritten in more readable way

File: player/src/main/java/com/focusit/jsflight/player/constants/BrowserType.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.focusit.jsflight.player.constants;
+
+/**
+ * Created by Gallyam Biktashev on 03.10.16.
+ */
+public enum BrowserType {
+    FIREFOX,
+    PHANTOM_JS,
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/constants/BrowserType.java]: Feel free to drop PHANTOM_JS. Unsupported browser at the time

File: player/src/main/java/com/focusit/jsflight/player/constants/EventType.java [hunk 3]
```diff
@@ -11,10 +11,10 @@
     public static final String CLICK = "click";
     public static final String KEY_UP = "keyup";
     public static final String KEY_DOWN = "keydown";
-    public static final String MOUSEDOWN = "mousedown";
+    public static final String MOUSE_DOWN = "mousedown";
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/constants/EventType.java:14]: Why have you used '_'?

File: server/src/main/java/com/focusit/jsflight/server/repository/EventRepository.java [hunk 4]
```diff
@@ -9,7 +9,7 @@
 /**
  * Created by dkirpichenkov on 20.05.16.
  */
-public interface EventRepositoryCustom
+public interface EventRepository
```
Comment (d0k1) [inline server/src/main/java/com/focusit/jsflight/server/repository/EventRepository.java:12]: I made 'custom' in sake of SpringData. I used Custom postfix to show that it's not  use any of spring.data magic

**Comment by d0k1** [comment]
And one more thing. Please be kind and merge all of your commits into one.

File: player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java (hunk 1 above)
Comment (gallyamb) [reply player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: This method have two options to implemented:
- reflection (as in this case)
- initalize each field manually by calling setter (which will look ugly)

Another approach, that will break `IDefaults` concept, is to annotate each field with custom annotation, e.g. `@DefaultScript(path="path/to/script")`, and then create this class instance via some factory and initialize it. But we still need reflection.

So, I think that current implementation is the most appropriate. It looks a bit complicated, but it's do his work perfectly. This code is readable and understandable

File: player/src/main/java/com/focusit/jsflight/player/constants/EventType.java (hunk 3 above)
Comment (gallyamb) [reply player/src/main/java/com/focusit/jsflight/player/constants/EventType.java:14]: For consistensy. Just like KEY_UP and KEY_DOWN

File: server/src/main/java/com/focusit/jsflight/server/repository/EventRepository.java (hunk 4 above)
Comment (gallyamb) [reply server/src/main/java/com/focusit/jsflight/server/repository/EventRepository.java:12]: Do we really need this postfix? I mean, that it's not so hard to look into implementation. But without postfix it look better, more consistently

File: player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java (hunk 1 above)
Comment (d0k1) [reply player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: Okay then, by how would I know about what script will be used to initialize a specific field.
what if I would change the field's name, what would happen then?

Comment (gallyamb) [reply player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: Well, now this piece of code should look better

Comment (d0k1) [reply player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: I see no difference. What am I  missing?

File: player/src/main/java/com/focusit/jsflight/player/constants/BrowserType.java (hunk 2 above)
Comment (gallyamb) [reply player/src/main/java/com/focusit/jsflight/player/constants/BrowserType.java]: What about chrome? We don't support it for now too

Comment (d0k1) [reply player/src/main/java/com/focusit/jsflight/player/constants/BrowserType.java]: Yep, but it isn't a trouble to support Chrome, at least I hope so. But it is almost impossible to support phantom_js, cause it is unsupported by selenium.

File: player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    }
+
+    public void syncScripts(JMeterRecorder recorder)
+    {
+        recorder.getScriptProcessor().setScenarioProcessorScript(getScenarioProcessorScript());
+        recorder.getScriptProcessor().setStepProcessorScript(getStepProcessorScript());
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    private @interface DefaultFile
```
Comment (d0k1) [inline player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: Usually annotation is declared in dedicated file with desired visibility level

Comment (gallyamb) [reply player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: Well, I planned, that this annotation is used only here, so there is no need to place it somewhere else. But, if it's required, of course, I can move it to another file

Comment (d0k1) [reply player/src/main/java/com/focusit/jsflight/player/configurations/ScriptsConfiguration.java]: I've never seen an inner annotation class.
Let's move it out.

**Pull request merged**